test/server-client-document-test.js

```diff
@@ -451,6 +451,144 @@ describe('Document style tests', function() {
 
   });
 
+  describe('Document/Literal wrapped with SOAP 1.2 Fault', function () {
+
+    var test = {};
+    test.server = null;
+    test.service = {
+      DocLiteralWrappedService: {
+        DocLiteralWrappedPort: {
+          myMethod: function (args, cb, soapHeader) {
+            throw {
+              Fault: {
+                Code: {
+                  Value: "soap:Sender",
+                  Subcode: { Value: "rpc:BadArguments" }
+                },
+                Reason: { Text: "Processing Error" },
+                Detail:
+                {myMethodFault2:
+                   {errorMessage2: 'MyMethod Business Exception message', value2: 10}
+                }
+              }
+            }
+          }
+        }
+      }
+    };
+
+    before(function (done) {
+      fs.readFile(__dirname + '/wsdl/strict/doc_literal_wrapped_test_soap12.wsdl', 'utf8', function (err, data) {
+        assert.ok(!err);
+        test.wsdl = data;
+        done();
+      });
+    });
+
+    beforeEach(function (done) {
+      test.server = http.createServer(function (req, res) {
+        res.statusCode = 404;
+        res.end();
+      });
+
+      test.server.listen(15099, null, null, function () {
```

Comment:
We probably should try to make the port number more env friendly to avoid conflicts on CI.